libs/utils/analysis/frequency_analysis.py

```diff
@@ -218,6 +218,107 @@ def plotClusterFrequencies(self, title='Clusters Frequencies'):
 
         return (avg_lfreq/1e3, avg_bfreq/1e3)
 
+    def plotCpusFrequencies(self, cpus=None):
+        """
+        Plot frequency trend for the specified CPUs (or all is not specified).
+        If sched_overutilized events ar available, the plots will also show the
+        intervals of time where the CPU was overutilized.
+
+        :param cpus: the list of CPUs to plot
+        :type cpus: int or list(int)
+        """
+        if not self._trace.hasEvents('cpu_frequency'):
+            self._log.warning('Events [cpu_frequency] not found, plot DISABLED!')
+            return
+        df = self._dfg_trace_event('cpu_frequency')
+
+        pd.options.mode.chained_assignment = None
+
+        freq = {}
+        for cpu_id in listify(cpus):
+            # Extract CPUs' frequencies and scale them to [MHz]
+            _df = df[df.cpu == cpu_id]
+            if _df.empty:
+                self._log.warning('No [cpu_frequency] events for CPU%d, plot DISABLED!', cpu)
+                continue
+            _df['frequency'] = _df['frequency']/1e3
+
+            # Compute AVG frequency for LITTLE cluster
+            avg_freq = 0
+            if len(_df) > 1:
+                _df['timestamp'] = _df.index
+                _df['delta'] = (_df['timestamp'] - _df['timestamp'].shift()).fillna(0).shift(-1)
+                _df['cfreq'] = (_df['frequency'] * _df['delta']).fillna(0)
+                timespan = _df.iloc[-1].timestamp - _df.iloc[0].timestamp
+                avg_freq = _df['cfreq'].sum()/timespan
```

Comment:
Can do this more simply using `bart.common.Utils.area_under_curve` [like in the load tracking test](https://github.com/ARM-software/lisa/blob/master/tests/eas/load_tracking.py#L142). Might get rid of the `chained_assignment` issue too?